storage/database/db_manager.go

```diff
@@ -1885,20 +1843,44 @@ func (dbm *databaseManager) ReadPreimageFromOld(hash common.Hash) []byte {
 	return data
 }
 
+func (dbm *databaseManager) WriteTrieNode(hash common.Hash, node []byte) error {
+	dbm.lockInMigration.RLock()
+	defer dbm.lockInMigration.RUnlock()
+
+	if dbm.inMigration {
+		if err := dbm.getDatabase(StateTrieMigrationDB).Put(TrieNodeKey(hash), node); err != nil {
+			return err
+		}
+	}
+	return dbm.getDatabase(StateTrieDB).Put(TrieNodeKey(hash), node)
+}
+
+func (dbm *databaseManager) PutTrieNodeToBatch(batch Batch, hash common.Hash, node []byte) error {
+	return batch.Put(TrieNodeKey(hash), node)
+}
+
 // WritePreimages writes the provided set of preimages to the database. `number` is the
 // current block number, and is used for debug messages only.
-func (dbm *databaseManager) WritePreimages(number uint64, preimages map[common.Hash][]byte) {
+func (dbm *databaseManager) WritePreimages(number uint64, preimages map[common.Hash][]byte) error {
 	batch := dbm.NewBatch(StateTrieDB)
 	for hash, preimage := range preimages {
 		if err := batch.Put(preimageKey(hash), preimage); err != nil {
```

Comment:
Don't we have to add `PutPreimageToBatch`?

Reply:
If there is such a use case, then yes.
But when I looked up the preimage-related code, they all simply use WritePreimages to write multiple preimages.

Reply:
Got it. Thanks.